ops/model.py

```diff
@@ -1790,6 +1790,14 @@ def __iter__(self):
     def __getitem__(self, key: Union['Unit', 'Application']) -> 'RelationDataContent':
         return self._data[key]
 
+    def __setitem__(self, key: Union['Unit', 'Application'], value: Mapping[str, str]) -> None:
+        try:
+            content = self._data[key]
+        except KeyError:
+            raise KeyError(f'Key "{key}" is not a known unit or app.') from None
```

Comment:
The message/str of a standard KeyError is `'a'` (including quotes). How about we kinda match that with the following:

```python
raise KeyError(f'{key!r} is not a known unit or app')
```